### PR TITLE
No longer restrict to Django<2

### PR DESCRIPTION
**Pull request by rmfitzpatrick**
From what I can tell, the current OpenTracingMiddleware implementation is Django 2+ compatible: https://docs.djangoproject.com/en/2.1/topics/http/middleware/

These changes remove the dependency restriction, add some specificity to the base tests, and update the necessary middleware setting per version.  I propose any discovered incompatibilities with newer Django versions are addressed as they arise instead of providing a hard limit.

I've successfully tested these changes with Django 1.9, 1.10, 1.11, 2.0 and 2.1.

edit: I've reverted the initial `OPENTRACING_TRACE_ALL` change when I realized that disables the middleware-triggered tracing.

edit 2: I've cherry-picked from https://github.com/opentracing-contrib/python-django/pull/33

### CODE REVIEW

**Comment by simonschmidt** [comment]
@rmfitzpatrick Feel free to move the tox changes in here :)

File: setup.py [hunk 1]
```diff
@@ -16,10 +16,13 @@
     packages=['django_opentracing', 'tests'],
     platforms='any',
     install_requires=[
-        'django<2',
+        'django',
```
Comment (crccheck) [inline setup.py]: I've found that life is easier if you don't put Django in `install_requires` at all. Similar to npm's peer dependencies concept. It won't work without Django, but no one is going to add `django_opentracing` to their `requirements.txt` and accept whatever version of Django it brings along. If there are version limitations, they can go into the documentation like [django-reversion](https://django-reversion.readthedocs.io/en/stable/django-versions.html)

Comment (rmfitzpatrick) [reply setup.py]: This seems not to be the OT-contrib convention for other instrumentations, but can potentially lead to problems.  @carlosalberto, do you have any preference or thoughts?

Comment (carlosalberto) [reply setup.py]: Well, usually in your application you have a dependency to `django` too (most likely with a specific version), and not a single dependency to `django_opentracing`.

On the other hand, I think we could put `django` in the test-requires section of `setup.py`, so the tests work just fine - but that would need to happen in a different PR/issue, so we can try it out (and later probably use this defensive behavior in other repos).

Makes sense?

**Comment by carlosalberto** [comment]
Going to close this in favor of #38 as this is simpler and already works with the upcoming `v1.0.0` branch - however, a few cleanups are needed still (a few updates from this PR, such as update for the example, additional assertions, README with updated comments, etc), so if you have time, it would be lovely to have them in a separated PR.

So sorry for the delayed review!

**Comment by spinus** [comment]
what's the status of this?
I'm asking as looks like the support for 2.0 was merged but readme still says about 1.0 limitation.

**Comment by rmfitzpatrick** [comment]
Looks like @carlosalberto removed the restriction on https://github.com/opentracing-contrib/python-django/tree/v1.0.0, but I've not had a chance to make the requested doc and test updates for that branch.  Closing this and will get those up asap.

**Comment by carlosalberto** [comment]
hey @spinus 

As @rmfitzpatrick mentioned, this has been fixed and released (both as `0.1.20`, which uses the `1.x` OT api, and `1.0.0`, which uses the OT 2.0 one).

I've updated the README to remove the limitation comment - had missed it ;)

**Comment by spinus** [comment]
great stuff, cheers.

**Pull request closed without merging**